include/jemalloc/internal/rb.h

```diff
@@ -852,6 +865,7 @@ a_prefix##remove(a_rbt_type *rbtree, a_type *node) {			\
 	    }								\
 	}								\
     }									\
+    rb_remove_safety_checks(rbtree, node, nodep, __func__);		\
```

Comment:
Sorry for the delay on the PR. As discussed we suspect this was caused by double free of sampled allocations -- let's add another sanity check before calling down to the remove function, to verify that the pointer being deallocated is still in an active state (not using the rb rtree, but rather its edata metadata).

This is probably a convenient place to do so: 
https://github.com/jemalloc/jemalloc/blob/86eb49b47847e48390c672371987ff4e476e53a3/include/jemalloc/internal/arena_inlines_b.h#L67

and we can add `large_dalloc_safety_checks` there, e.g.
```
if (reset_recent) {
    large_dalloc_safety_checks(edata, ptr, edata_szind_get(edata));
}
```
It's not ideal to do this in a _info_get utility function, but this looks to have all the relevant context, with the `reset_recent` flag indicating a one-time operation that can only be done on active allocations.